plugins/auth-node/src/IdentityClient.ts

```diff
@@ -13,22 +13,32 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 import { PluginEndpointDiscovery } from '@backstage/backend-common';
 import { AuthenticationError } from '@backstage/errors';
 import {
   createRemoteJWKSet,
   decodeJwt,
-  jwtVerify,
+  decodeProtectedHeader,
   FlattenedJWSInput,
   JWSHeaderParameters,
-  decodeProtectedHeader,
+  jwtVerify,
 } from 'jose';
 import { GetKeyFunction } from 'jose/dist/types/types';
+
 import { BackstageIdentityResponse } from './types';
 
 const CLOCK_MARGIN_S = 10;
 
+export type IdentityClientOptions = {
+  discovery: PluginEndpointDiscovery;
+  issuer: string;
+
+  /** JWS "alg" (Algorithm) Header Parameter value. Defaults to ES256.
+   * Must match the algorithm defined in TokenFactory.
+   * More info on supported algorithms: https://github.com/panva/jose */
+  algorithm?: string;
```

Comment:
I think it would make sense to make this option an array upfront, defaulting it to `['ES256']`. We'll need that for migrations and whatnot in the future anyway.

Reply:
I was in doubt before going for the `algorithm` string instead of `algorithms` array.
 
So, is IdentityClient going to have an array of algorithms, while TokenFactory not? 
To me, both should have an array of algorithms but then how TokenFactory is supposed to select one of them?